RecoLocalTracker/SiPixelRecHits/plugins/PixelCPEGenericESProducer.cc

```diff
@@ -98,4 +96,25 @@ std::unique_ptr<PixelClusterParameterEstimator> PixelCPEGenericESProducer::produ
                                            lorentzAngleWidthProduct);
 }
 
+void PixelCPEGenericESProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+
+  // from PixelCPEBase
+  PixelCPEBase::fillPSetDescription(desc);
+
+  // from PixelCPEGeneric
+  PixelCPEGeneric::fillPSetDescription(desc);
+
+  // specific to PixelCPEGenericESProducer
+  desc.add<std::string>("ComponentName", "PixelCPEGeneric");
+  desc.add<edm::ESInputTag>("MagneticFieldRecord", edm::ESInputTag(""));
+  desc.add<bool>("useLAAlignmentOffsets", false);
+
+  // vestigial (present in certain HLT menus)
+  desc.addOptional<double>("TanLorentzAnglePerTesla", 0.106);
```

Comment:
Since you are giving a default value, there is no need for making it `optional`.

Reply:
Or is it optional because the module no longer uses that parameter? If  that is the case, you can just drop the default as it doesn't do anything.

Reply:
@Dr15Jones the problem is that this parameter is indeed not used in the module, but it is present in some of HLT configuration, that as far as I know I cannot modify, and if I remove it, it complains that an illegal parameter is used.

Reply:
I'm sorry I wasn't clear my suggestion is to do
```suggestion
  desc.addOptional<double>("TanLorentzAnglePerTesla");
```
which will allow the value to be defined in the ParameterSet but will not inject a value if it is not there.

Reply:
thanks! I was not aware of this possibility. It's committed now.